osu.Game/Database/UserLookupCache.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Users;
+
+namespace osu.Game.Database
+{
+    public class UserLookupCache : MemoryCachingComponent<int, User>
+    {
+        private readonly ConcurrentBag<int> nextTaskIDs = new ConcurrentBag<int>();
```

Comment:
I'm a bit confused as to why this is a concurrent collection, as all accesses to this happen only if `taskAssignmentLock` is taken...

Reply:
Aware of that, left it due to the better API of the class (and because it's such a tiny use case that wouldn't affect anything). But based on it containing duplicates I'll change away.